flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1599,6 +1603,33 @@ private OptionalLong restoreLatestCheckpointedStateInternal(
         }
     }
 
+    private Map<OperatorID, OperatorState> extractOperatorStates(CompletedCheckpoint checkpoint) {
+        Map<OperatorID, OperatorState> operatorStates = checkpoint.getOperatorStates();
```

Comment:
I think we should NOT modify the checkpoint object in memory.
Otherwise, channel state will not be discarded later when the checkpoint is subsumed.

Reply:
Yes, it makes sense to create a new map rather than modify an existing one.